BaseWeapon/src/main/java/com/github/sef24sp4/baseweapon/BaseWeapon.java

```diff
@@ -0,0 +1,61 @@
+package com.github.sef24sp4.baseweapon;
+
+import com.github.sef24sp4.common.entities.ICollidableEntity;
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IEntityManager;
+import com.github.sef24sp4.common.weapon.WeaponSPI;
+
+public class BaseWeapon implements WeaponSPI {
+	private int ammoCount;
+	private final int maxAmmoCount;
```

Comment:
Because this weapon has infinite ammunition, only a single private final field is needed. Please remove redundant field.